### DISCUSSION TITLE
RESTEasy Reactive: URI mapping issue

### DISCUSSION THREAD

**Discussion by jrongs** [quarkusio/quarkus#32098] [Q&A]
Hello,

Today we encountered a rather interesting URI mapping issue, lets say we have the following simple resource class:

```java
@Path("/hello")
public class GreetingResource {

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    @Path("/c/{id}")
    public String cPathId(@PathParam("id") String id) {
        return "c path id: " + id;
    }

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    @Path("/{id}")
    public String pathId(@PathParam("id") String id) {
        return "path id: " + id;
    }
}
```
> curl localhost:8080/hello/c/123    ->  c path id: 123
> curl localhost:8080/hello/c123      -> * HTTP ERROR 404 *
> curl localhost:8080/hello/b123     -> path id: b123

I was expecting  /hello/c123 to match the /{id} path, but  getting a 404 instead.  looks like it is because the requested path starts with /c, which partially match the /c/{id} path, and that caused a mapping failure.   nothing in the dev console log and checked against all recent quarkus releases.   is this a bug or something wrong with my mapping above?  any clarification on this will be greatly appreciated.  

Thanks!

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
What version of Quarkus are you seeing this problem with?

**Reply by jrongs** [reply]
@geoand  thanks for looking into this issue.  we are on Quarkus version 2.16.5, but replicated the same issue on all 2.16.x and 2.15.x.  versions. Thanks!

**Reply by geoand** [reply]
Thanks for checking.

This sounds like a bug to be honest

**Comment by stuartwdouglas** [comment]
This is a bug

**Comment by jrongs** [comment]
@geoand  @stuartwdouglas  thanks for your quick responses, I'll open a bug report on this.  thanks!
